codec_test.go

```diff
@@ -940,10 +983,15 @@ func Test_DecodeProtocolMessage(t *testing.T) {
 		{loadedSourceEventString, &loadedSourceEventStruct, noError},
 		{processEventString, &processEventStruct, noError},
 		{capabilitiesEventString, &capabilitiesEventStruct, noError},
+		{customEventString, &customEventStruct, noError},
 	}
+
+	codec := NewCodec()
+	codec.RegisterRequest("customReq", func() Message { return new(customRequest) }, func() Message { return new(customResponse) })
+	codec.RegisterEvent("customEvt", func() Message { return new(customEvent) })
 	for _, test := range tests {
 		t.Run(test.data, func(t *testing.T) {
-			msg, err := DecodeProtocolMessage([]byte(test.data))
+			msg, err := codec.DecodeMessage([]byte(test.data))
```

Comment:
Thanks for updating the test! I think we should keep the `DecodeProtocolMessage` as the main function that we test.

We could instead test each of these test cases on both `codec.DecodeMessage` and `DecodeProtocolMessage` (except the custom events which should only succeed on `codec.DecodeMessage`). 

Maybe something like:
```
for _, decoder := range []func([]byte) (Message, error){DecodeProtocolMessage, codec.DecodeProtocolMessage} {
  for _, test := range tests {
      t.Run(test.data, func(t *testing.T) {
          ...
      }
  }
}
```

The two custom cases could either be split into their own test loop or have another field added to the test case to distinguish them. 

What do you think? Another option might be to test codec.DecodeMessage on a smaller subset if it seems better to break it into a separate test instead.

Reply:
Yes, that makes sense. I added a new test function for this. PTAL